### PR TITLE
Adding functions to interacting with db

### PR DESCRIPTION
**Pull request by AymericNoel**
I have added functions to interact with database for the transactions.
These functions are useful for my next pull requests.
For the tests, I've taken the previous advice, no tests overlap 🥇

### CODE REVIEW

File: src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        public async Task GetAllIndiceSupplyTransactionsByUser_should_return_empty_list_if_user_didnt_make_transaction()
+        {
+            var result = await _transactionDataProvider.GetAllWrappingTransactionsByUser("userTest12/05/2020;09h58;125seconds");
+            result.Should().BeEmpty();
+        }
+
+        [Fact]
+        public async Task GetAllWrappingTransactionsByUser_should_return_list_of_wrappingTransaction()
+        {
+            var transaction = await PutWrappingTransactionInDatabase();
```
[the remark this answers is not in the document]
Comment (monsieurleberre) [reply src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs:89]:
```suggestion
            var transaction = await PutWrappingTransactionInDatabase().ConfigureAwait(false);
```

File: src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                await _transactionDataProvider.GetWrappingTransactionByNativeTransactionHash(
+                    "test12/05/2020;09h43;000236987secondes");
+
+            result.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task GetWrappingTransactionByEthereumTransactionHash_should_return_the_transaction()
+        {
+            var transaction = await PutWrappingTransactionInDatabase();
```
[the remark this answers is not in the document]
Comment (AymericNoel) [reply src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs:62]:
```suggestion
            var transaction = await PutWrappingTransactionInDatabase().ConfigureAwait(false);;
```

File: src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                await _transactionDataProvider.GetWrappingTransactionByNativeTransactionHash(
+                    "test12/05/2020;09h33;16515612secondes");
+
+            result.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task GetWrappingTransactionByNativeTransactionHash_should_return_the_transaction()
+        {
+            var transaction = await PutWrappingTransactionInDatabase();
```
[the remark this answers is not in the document]
Comment (jmc171144) [reply src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs:38]:
```suggestion
            var transaction = await PutWrappingTransactionInDatabase().ConfigureAwait(false);
```

File: src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            result.SenderAddress.Should().Be(transaction.SenderAddress);
+            result.Amount.Should().Be(transaction.Amount);
+            result.TimeStamp.Should().Be(transaction.TimeStamp);
+            result.ReceiverAddress.Should().Be(transaction.ReceiverAddress);
+            result.NativeChainTransactionHash.Should().Be(transaction.NativeChainTransactionHash);
+        }
+
+        public async Task<WrappingTransactionDao> PutWrappingTransactionInDatabase()
+        {
+            var transaction = new WrappingTransactionDao(new DateTime(2020, 12, 3), "xrp", "wxrp", TransactionState.Pending,
```
Comment (monsieurleberre) [inline src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs:135]: Cool I love the intention, but maybe it is easier to have this transaction a bit randomised and then you don't need to remove the record and save changes at each test 🤔
If you forget to remove in one test, then you don't know if the others are actually testing what you want, so it is better to try to keep each test independent of each other.

File: src/Trakx.Tests/Unit/Persistence/TransactionDataCreatorTest.cs [hunk 5]
```diff
@@ -40,7 +40,8 @@ public async Task SaveIndiceSupplyTransaction_should_return_true_if_addition_suc
                 .Be(transactionToSave.IndiceCompositionDao.IndiceDefinitionDao.Name);
             retrievedTransaction.Quantity.Should().Be(transactionToSave.Quantity);
 
-                _context.IndiceSupplyTransactions.Remove(transactionToSave);
+            _context.IndiceSupplyTransactions.Remove(transactionToSave);
```
Comment (monsieurleberre) [inline src/Trakx.Tests/Unit/Persistence/TransactionDataCreatorTest.cs:43]: still needed?

File: src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs [hunk 6]
```diff
@@ -130,15 +140,5 @@ public async Task GetAllIndiceSupplyTransactionsByUser_should_return_list_of_ind
             result.NativeChainTransactionHash.Should().Be(transaction.NativeChainTransactionHash);
         }
```
Comment (monsieurleberre) [inline src/Trakx.Tests/Unit/Persistence/TransactionDataProviderTest.cs:142]: No 😲 it doesn't make sense to remove that method and copy its content three times in the file!!

**Pull request merged**